Makefile

```diff
@@ -0,0 +1,35 @@
+.PHONY: all configure build repl nix-shell update-nix
+
+all:
+	-@echo "Targets:  update-nix"
+	-@echo "  configure:  cabal configure"
+	-@echo "  build:      cabal build"
+	-@echo "  repl:       cabal repl"
+	-@echo "  nix-build:  build via nix"
+	-@echo "  nix-shell:  open a nix shell"
+	-@echo "  update-nix: update the nix files"
+
+configure:
+	cabal configure
+
+build:
+	cabal build -j
+
+repl:
+	cabal repl
+
+nix-build: nix/capnp.nix
+	-@rm -f result*
+	nix-build --no-out-link shell.nix -Q -j 8
+
+nix-shell: nix/capnp.nix
+	cabal clean
```

Comment:
Can you explain to me why cabal clean is needed here? I must be missing something.

Reply:
I've had problems with `dist` breaking `cabal build` after entering a nix shell. Not entirely sure why they happen (though I have some theories), but they are always fixed by `cabal clean`ing before entering the nix shell.

Reply:
OK -- would you mind adding a comment explaining that (and maybe your hypothesis as well?)